src/app/reporting/EventLoggingTypes.h

```diff
@@ -0,0 +1,227 @@
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file
+ *
+ * @brief
+ *   Enums, types, and tags used in Chip Event Logging.
+ *
+ */
+
+#pragma once
+
+#include <core/CHIPCore.h>
+#include <core/CHIPTLV.h>
+#include <system/SystemPacketBuffer.h>
+#include <util/basic-types.h>
+
+namespace chip {
+namespace app {
+namespace reporting {
+
+/**
+ * @brief
+ *   The Priority of the log entry.
+ *
+ * @details
+ * Priority is used as a way to filter events before they are
+ * actually emitted into the log. After the event is in the log, we
+ * make no further provisions to expunge it from the log.
+ * The priority level serves to prioritize event storage. If an
+ * event of high priority is added to a full buffer, events are
+ * dropped in order of priority (and age) to accommodate it. As such,
+ * priority levels only have relative value. If a system is
+ * using only one priority level, events are dropped only in order
+ * of age, like a ring buffer.
+ */
+
+typedef enum PriorityLevel
+{
+
+    kPriorityLevel_First = 1,
+    /**
+     * Critical priority denotes events whose loss would
+     * directly impact customer-facing features. Applications may use
+     * loss of Production Critical events to indicate system failure.
+     * On constrained devices, entries logged with Critical
+     * priority must be accounted for in the power and memory budget,
+     * as it is expected that they are always logged and offloaded
+     * from the device.
+     */
+    Critical = kPriorityLevel_First,
+    /**
+     * Info priority denotes log entries that provide extra insight
+     * and diagnostics into the running system. Info logging level may
+     * be used over an extended period of time in a production system,
+     * or may be used as the default log level in a field trial. On
+     * the constrained devices, the entries logged with Info level must
+     * be accounted for in the bandwidth and memory budget, but not in
+     * the power budget.
+     */
+    Info = 2,
+
+    /**
+     *  Debug priority denotes log entries of interest to the
+     *  developers of the system and is used primarily in the
+     *  development phase. Debug priority logs are
+     *  not accounted for in the bandwidth or power budgets of the
+     *  constrained devices; as a result, they must be used only over
+     *  a limited time span in production systems.
+     */
+    Debug                  = 3,
+    kPriorityLevel_Last    = Debug,
+    kPriorityLevel_Invalid = kPriorityLevel_Last + 1,
+
+} PriorityLevel;
+
+/**
+ * @brief
+ *   The structure that defines a schema for event metadata.
+ */
+struct EventSchema
+{
+    chip::ClusterId mClusterId; //< ID of cluster
```

Comment:
I'm not quite sure why this is part of `EventSchema` while `EventOptions` is where the node/endpoint are stored.  Why the difference?

Reply:
I would update EventSchema, and put the info inside EventOptions.